src/mlpack/methods/dbscan/dbscan_main.cpp

```diff
@@ -86,26 +88,48 @@ void RunDBSCAN(RangeSearchType rs = RangeSearchType())
 
   // Load dataset.
   arma::mat dataset = std::move(CLI::GetParam<arma::mat>("input"));
-
   const double epsilon = CLI::GetParam<double>("epsilon");
   const size_t minSize = (size_t) CLI::GetParam<int>("min_size");
-
-  DBSCAN<RangeSearchType> d(epsilon, minSize, !CLI::HasParam("single_mode"),
-      rs);
-
-  // If possible, avoid the overhead of calculating centroids.
   arma::Row<size_t> assignments;
-  if (CLI::HasParam("centroids"))
-  {
-    arma::mat centroids;
-
-    d.Cluster(dataset, assignments, centroids);
 
-    CLI::GetParam<arma::mat>("centroids") = std::move(centroids);
+  // Check if random selection is used.
+  if (CLI::HasParam("random_selection")) // Random Selection
+  {
+    DBSCAN<RangeSearchType, RandomPointSelection> d(epsilon, minSize, 
+        !CLI::HasParam("single_mode"), rs);
+
+    // If possible, avoid the overhead of calculating centroids.
+    if (CLI::HasParam("centroids"))
+    {
+      arma::mat centroids;
+
+      d.Cluster(dataset, assignments, centroids);
+
+      CLI::GetParam<arma::mat>("centroids") = std::move(centroids);
+    }
+    else
+    {
+      d.Cluster(dataset, assignments);
+    }
   }
-  else
+  else // Ordered Selection
   {
-    d.Cluster(dataset, assignments);
+    DBSCAN<RangeSearchType> d(epsilon, minSize,
+        !CLI::HasParam("single_mode"), rs);
+        
```

Comment:
Since only the call `DBSCAN<RangeSearchType, ...>` has to be different, I think we can save a bit on code duplication here.

Reply:
I totally agree. I will make new commit.
Could you please review it later??

Reply:
On a second thought, I feel code deduplication will have to come at a cost of using the assignment operation, and in that case it's better to leave it as it is, if only two policies ae going to require the code below. @rcurtin Is there a possibility that another policy might require the same piece of code as below? I'm not sure how many different selection policies are developed for DBSCAN. If they are indeed numerous, we can implement the below as a function.

Reply:
@KimSangYeon-DGU I don't think you ever need to request for reviews. For every commit you push or comment you make, we get notification emails since we're watching this repository. And we do check our mail :) Only in case where you PR has been left unchecked for let's say more than 3 days, you should hit one of us up.

Reply:
> On a second thought, I feel code deduplication will have to come at a cost of using the assignment operation, and in that case it's better to leave it as it is, if only two policies are going to require the code below. @rcurtin Is there a possibility that another policy might require the same piece of code as below? I'm not sure how many different selection policies are developed for DBSCAN. If they are indeed numerous, we can implement the below as a function.

First of all, I've modified to reflect your feedback for extensibility.

Reply:
I see, I like the current changes, they're minimal.